### PR TITLE
Observability jmx

### PR DESCRIPTION
**Pull request by bonclay7**
### What does this PR do?

This PR provides patterns for Observability on EKS with provided metrics collection with OpenTelemetry and default dashboards on Grafana

Rework of https://github.com/aws-ia/terraform-aws-eks-blueprints/pull/347

### More

- [x] Yes, I have tested the PR using my local account setup  (Provide any test evidence report under Additional Notes)
- [x] Yes, I have added a new example under [examples](https://github.com/aws-samples/aws-eks-accelerator-for-terraform/tree/main/examples) to support my PR
- [ ] Yes, I have created another PR for add-ons under [add-ons](https://github.com/aws-samples/ssp-eks-add-ons) repo (if applicable)
- [x] Yes, I have updated the [docs](https://github.com/aws-samples/aws-eks-accelerator-for-terraform/tree/main/docs) for this feature
- [x] Yes, I ran `pre-commit run -a` with this PR

**Note**: Not all the PRs required examples and docs except a new pattern or add-on added.

### For Moderators
- [x] E2E Test successfully complete before merge?

### CODE REVIEW

File: examples/observability/eks-java-jmx/data.tf [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+data "aws_availability_zones" "available" {}
+
+data "aws_eks_cluster" "cluster" {
+  name = module.aws-eks-accelerator-for-terraform.eks_cluster_id
+}
+
+data "aws_eks_cluster_auth" "cluster" {
+  name = module.aws-eks-accelerator-for-terraform.eks_cluster_id
+}
```
Comment (vara-bonthu) [inline examples/observability/eks-java-jmx/data.tf:11]: these can go into `main.tf` file. We would like to keep one file for examples to keep ti simple

File: examples/observability/eks-java-jmx/dev.tfvars [hunk 2]
```diff
@@ -0,0 +1,2 @@
+grafana_api_key  = "<inset grafana api key>"
+grafana_endpoint = "https://<grafana endpoint>/"
```
Comment (vara-bonthu) [inline examples/observability/eks-java-jmx/dev.tfvars:2]: Add this to deployment README on what customers needs to do for this variable

File: examples/observability/eks-java-jmx/providers.tf [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    host                   = data.aws_eks_cluster.cluster.endpoint
+    token                  = data.aws_eks_cluster_auth.cluster.token
+    cluster_ca_certificate = base64decode(data.aws_eks_cluster.cluster.certificate_authority.0.data)
+  }
+}
+
+provider "grafana" {
+  url  = var.grafana_endpoint
+  auth = var.grafana_api_key
+}
```
Comment (vara-bonthu) [inline examples/observability/eks-java-jmx/providers.tf:23]: move this to `main.tf`

File: examples/observability/eks-java-jmx/variables.tf [hunk 4]
```diff
@@ -0,0 +1,8 @@
+# variable "grafana_workspace_id" {}
+variable "grafana_endpoint" {}
```
Comment (vara-bonthu) [inline examples/observability/eks-java-jmx/variables.tf:2]: Add `type` and `description`

File: examples/observability/eks-java-jmx/variables.tf [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+
+variable "grafana_api_key" {
+  type        = string
+  sensitive   = true
+  description = "Api key for authorizing the Grafana provider to make changes to Amazon Managed Grafana"
+}
```
Comment (vara-bonthu) [inline examples/observability/eks-java-jmx/variables.tf:8]: You can just these two variables as expected and remove the `tfvars` file to further simplify

File: modules/kubernetes-addons/aws-observability-pattern-jmx/dashboards/default.json [hunk 6]
```diff
@@ -0,0 +1,1790 @@
...
+          "limit": 100,
+          "matchAny": false,
+          "tags": [],
+          "type": "dashboard"
+        },
+        "type": "dashboard"
+      }
+    ]
+  },
+  "description": "Dashboard for JVM metrics with Prometheus / JMX Exporter",
```
Comment (vara-bonthu) [inline modules/kubernetes-addons/aws-observability-pattern-jmx/dashboards/default.json:21]: Should the Dashboard be in examples?

**Review by vara-bonthu — COMMENTED** [review]
Thanks @bonclay7! Minor comments

We just to run E2E on this before the merge

File: examples/observability/eks-java-jmx/dev.tfvars (hunk 2 above)
Comment (saaish) [reply examples/observability/eks-java-jmx/dev.tfvars:2]: added

File: examples/observability/eks-java-jmx/providers.tf (hunk 3 above)
Comment (saaish) [reply examples/observability/eks-java-jmx/providers.tf:23]: Moved

File: modules/kubernetes-addons/adot-collector-java/locals.tf [hunk 7]
```diff
@@ -0,0 +1,60 @@
+locals {
+  name                           = "adot-collector-java"
+  adot_collector_service_account = "adot-collector-java"
+
+  default_helm_config = {
+    name        = local.name
+    repository  = null
+    chart       = "${path.module}/otel-config"
```
Comment (bryantbiggs) [inline modules/kubernetes-addons/adot-collector-java/locals.tf:8]: I didn't do a full comparison but just curious why the local/custom helm chart and not the one provided by opentelemetry? https://github.com/open-telemetry/opentelemetry-helm-charts/tree/main/charts

**Review by bryantbiggs — APPROVED** [review]
Looks good enough to ship

**Review by Zvikan — APPROVED** [review]
LGTM assuming e2e/ local tests ran successfully! ty

**Pull request merged**